packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1103,6 +1104,45 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   /// Override in subclasses with children and call the visitor for each child.
   void visitChildren(RenderObjectVisitor visitor) { }
 
+  /// Capture an image of the current state of the render object and its
+  /// children.
+  ///
+  /// The returned [ui.Image] has uncompressed raw bytes in the dimensions of
+  /// the render object, multiplied by the [window.devicePixelRatio].
+  ///
+  /// The property [isRepaintBoundary] must be true for [toImage] to work, and
+  /// the render object must not need layout, painting, or compositing.
+  ///
+  /// The [pixelRatio] describes the scale between the logical pixels and the
+  /// size of the output image. It is independent of the
+  /// [window.devicePixelRatio] for the device, so specifying 1.0 (the default)
+  /// will give you a 1:1 mapping between logical pixels and the output pixels
+  /// in the image. The output image is not anti-aliased, so you may wish to
```

Comment:
Where is anti-aliasing usually done when the render object is drawn to the screen?

Reply:
I don't really understand how we're actually rendering this, but I would presume we're just uploading the whole scene to the GPU, having it render to a texture, then grabbing the texture? In which case I'd expect antialiasing to work fine?

cc @jason-simmons or @chinmaygarde

Reply:
Yeah, what @Hixie said. We have to render to texture on the GPU first because the scene may reference images that are only resident on the GPU. AA should work just fine.